server/controllers/classifier.js

```diff
@@ -168,55 +192,57 @@ class Classifier {
     /**
      * classifies a token with levenshtein distance
      * @param labelInfos possible values of the category
-     * @param lookahead false if no lookahead should be performed
      */
-    classifyToken(labelInfos, lookahead) {
+    classifyToken(labelInfos) {
         let tokenHolder = this.state.tokenHolders[this.state.currentToken];
-        let performedLookahead = false;
-        let labeledTokenInfo = Classifier.getMostLikelyMatch(tokenHolder.token, labelInfos);
-        if (labeledTokenInfo && labeledTokenInfo.distance < tokenHolder.distance) {
+        let labeledTokenInfo = this.getMostLikelyMatch(labelInfos);
+        if (labeledTokenInfo && labeledTokenInfo.distance <= tokenHolder.distance && (tokenHolder.label == null || labeledTokenInfo.label.length >= tokenHolder.label.length)) {
             // this layer matches for this token
             this.state.tokenHolders[this.state.currentToken] = labeledTokenInfo;
-        } else if (lookahead && this.state.currentToken <= this.state.tokenHolders.length - 2) {
-            let nextTokenHolder = this.state.tokenHolders[this.state.currentToken + 1];
-            labeledTokenInfo = Classifier.getMostLikelyMatch(tokenHolder.token + " " + nextTokenHolder.token, labelInfos);
-            if (labeledTokenInfo && labeledTokenInfo.distance < tokenHolder.distance) {
-                performedLookahead = true;
-                //replace the currentToken
-                this.state.tokenHolders[this.state.currentToken] = labeledTokenInfo;
-            }
+
         }
-        this.nextAction(performedLookahead);
+        this.nextAction();
     }
 
     /**
      * gets the most likely match for a token out of an array of values
-     * @param token token to match
      * @param labelInfos possible values which it could match
      * @returns {*}
```

Comment:
maybe add the documentation for the new return

Reply:
done